### PR TITLE
creating digestfile directory before writing

### LINKED ISSUE #813: The /kaniko/executor --digest-file option should create parent directories
**Actual behavior**
I run Kaniko Docker container with the --digest-file option:
```
/kaniko/executor ... --digest-file /tmp/outputs/Image_name_with_digest/data
```
The execution fails due to the lack of parent directories:
```
[33mWARN[0m[0055] error uploading layer to cache: writing digest to file failed: open /tmp/outputs/Image_name_with_digest/data: no such file or directory 
error pushing image: writing digest to file failed: open /tmp/outputs/Image_name_with_digest/data: no such file or directory
```

I cannot add a script to pre-create the parent directories, because Kaniko image is distroless and lacks `sh`.

**Expected behavior**
I expect Kaniko to create the parent directories for any output file it writes.

 | **Description** | **Yes/No** |
 |----------------|---------------|
 | Please check if this a new feature you are proposing        | <ul><li>- [ ] </li></ul>|
 | Please check if the build works in docker but not in kaniko | <ul><li>- [ ] </li></ul>| 
 | Please check if this error is seen when you use `--cache` flag | <ul><li>- [ ] </li></ul>|
 | Please check if your dockerfile is a multistage dockerfile | <ul><li>- [ ] </li></ul>|

### PR DESCRIPTION
**Pull request by prary**
Fixes #813 . 

**Description**

creating digestfile directory before writing

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [ ] Adds integration tests if needed.

_See [the contribution guide](../CONTRIBUTING.md) for more details._

**Reviewer Notes**

- [ ] The code flow looks good. 
- [ ] Unit tests and or integration tests added.

### CODE REVIEW

File: pkg/executor/push.go [hunk 1]
```diff
@@ -109,6 +109,13 @@ func DoPush(image v1.Image, opts *config.KanikoOptions) error {
 	t := timing.Start("Total Push Time")
 	var digestByteArray []byte
 	var builder strings.Builder
+
+	dir := strings.Split(opts.DigestFile, "/")
+	parentDir := strings.Join(dir[:len(dir)-1], "/")
```
Comment (cvgw) [inline pkg/executor/push.go]: Lines 113 and 114 can be replaced with `parentDir := filepath.Dir(opts.DigestFile)`

File: pkg/executor/push.go [hunk 2]
```diff
@@ -109,6 +109,13 @@ func DoPush(image v1.Image, opts *config.KanikoOptions) error {
...
+	if _, err := os.Stat(parentDir); os.IsNotExist(err) {
+		os.MkdirAll(parentDir, 0700)
+		logrus.Infof("Created directory %v", parentDir)
```
Comment (cvgw) [inline pkg/executor/push.go]: Nit: Debug or Trace level

**Comment by cvgw** [comment]
rebase against master

File: pkg/executor/push.go (hunk 1 above)
Comment (prary) [reply pkg/executor/push.go]: updated

**Comment by cvgw** [comment]
@prary looks like there are some linting errors that need to be fixed for travis to pass

**Comment by prary** [comment]
@cvgw updated the PR now travis is passing

File: pkg/executor/push.go [hunk 3]
```diff
@@ -109,6 +109,12 @@ func DoPush(image v1.Image, opts *config.KanikoOptions) error {
 	t := timing.Start("Total Push Time")
 	var digestByteArray []byte
 	var builder strings.Builder
+
+	parentDir := filepath.Dir(opts.DigestFile)
```
Comment (tejal29) [inline pkg/executor/push.go]: Can we move this entire logic into a new function `writeDigestFile(string, []byte)`
```
func writeDigestFile(path string, digestByteArray []byte) error {
    if _, err := os.Stat(filepath.Dir(path)); os.IsNotExist(err) {
	        if err := os.MkdirAll(parentDir, 0700); err != nil {
		  logrus.Debugf("error creating %s, %s", parentDir, err)
                }
                logrus.Tracef("Created directory %v", parentDir)
   }
   return ioutil.WriteFile(opts.DigestFile, digestByteArray, 0644
}
```

Comment (prary) [reply pkg/executor/push.go]: Sure will update and add unit tests as well :)

**Comment by tejal29** [comment]
@prary are you still working on this?

**Comment by tejal29** [comment]
closing due to inactivity

**Pull request closed without merging**